### ISSUE TITLE
Rule change: no-extra-boolean-cast should catch more cases

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#12137]
**What rule do you want to change?**

[no-extra-boolean-cast](https://eslint.org/docs/rules/no-extra-boolean-cast)

**Does this change cause the rule to produce more or fewer warnings?**

More

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior

**Please provide some example code that this change will affect:**

```js
if (foo && !!bar) {}
if (!!foo && bar) {}

while (foo && bar && !!baz) {}

if (foo && Boolean(bar) {}
```

**What does the rule currently do for this code?**

No errors

**What will the rule do after it's changed?**

Throw a `Redundant double negation` error.

**Are you willing to submit a pull request to implement this change?**

Possibly, depends what time I get.

**Comment by mdjermanovic** [comment]
:thumbsup: from me for the enhancement in general.

If accepted, this should probably be an option rather than default behavior, until the next major release.

The same could apply to `||` as well?

**Comment by g-plane** [comment]
I prefer this is as default behavior personally.

**Comment by Standard8** [comment]
I guess if it is default immediately or not, depends on if it is seen as breaking behaviour?

**Comment by platinumazure** [comment]
As a reminder, we do need a team member to champion this rule proposal before we can mark this issue as accepted. Would anyone from the team like to champion this rule?

Regarding default behavior or not: If the proposal here is to change the default behavior, then this is a breaking change and a PR cannot be merged until the next major release. If we want the functionality available before then, we need to have this behavior behind an option with the default behavior unchanged.

**Comment by mdjermanovic** [comment]
I'll champion this and there are already 3 other :thumbsup:, so marked as accepted!

This feature has to be behind an option at the moment. The option applies to both  `&&` and `||`.

Any ideas for the name, maybe `enforceForLogicalOperands`?

@Standard8 would you like to submit a PR to implement this option?

**Comment by Standard8** [comment]
> @Standard8 would you like to submit a PR to implement this option?

I'll be happy to, but realistically, it is going to be a couple of weeks before I get time to write it.

**Comment by mdjermanovic** [comment]
> I'll be happy to, but realistically, it is going to be a couple of weeks before I get time to write it.

Sounds good to me! It's perfectly fine whenever you get time.

**Comment by jmoore914** [comment]
If @Standard8 doesn't have the time to work on this, I would be happy to!

**Comment by Standard8** [comment]
@jmoore914 Please do, I'm still struggling to spare the time for this.
